controllers/release/release_adapter.go

```diff
@@ -582,13 +582,15 @@ func (a *Adapter) registerGitOpsDeploymentStatus(binding *applicationapiv1alpha1
 		patch := client.MergeFrom(a.release.DeepCopy())
 
 		condition := meta.FindStatusCondition(binding.Status.ComponentDeploymentConditions, v1alpha1.BindingDeploymentStatusConditionType)
-		if condition.Status == metav1.ConditionUnknown {
-			a.release.MarkDeploying(condition.Reason, condition.Message)
-		} else {
-			a.release.MarkDeployed(condition.Status, condition.Reason, condition.Message)
-		}
+		if condition != nil {
+			if condition.Status == metav1.ConditionUnknown {
+				a.release.MarkDeploying(condition.Reason, condition.Message)
+			} else {
+				a.release.MarkDeployed(condition.Status, condition.Reason, condition.Message)
+			}
 
-		return a.client.Status().Patch(a.context, a.release, patch)
+			return a.client.Status().Patch(a.context, a.release, patch)
+		}
```

Comment:
Too many levels. Let's take the chance to change the function to something like:
```
if binding == nil
    return nil

condition := ...

if condition == nil
    return nil
    
if condition.Status ... {
}

return a.client.Status...
```